lib/src/kvs/clock.rs

```diff
@@ -0,0 +1,80 @@
+use crate::dbs::node::Timestamp;
+use crate::sql;
+use sql::Duration;
+use std::sync::Arc;
+use std::time::{SystemTime, UNIX_EPOCH};
+use tokio::sync::Mutex;
+
+// Traits cannot have async and we need sized structs for Clone + Send + Sync
```

Comment:
@phughk Just confirming; Is this the reason to introduce another clock? 😺

Reply:
Sadly yes 😂 More importantly, it is why we are exposing test clocks in the release code. It is what it is unfortunately.

Reply:
I will reconcile all the clocks to a single implementation though (including the versionstamp code)

Reply:
I come late in the topic, but if this is the reason, would this approach be an option `Pin<Box<dyn Future....` ?

https://github.com/surrealdb/surrealdb/blob/74ad6937095e1cab33648c47a3147c49ce6393a2/lib/src/api/conn.rs#L169-L172

Reply:
There are 2 issues:
1. We have a lot of clock implementations. CF adds 2 (versionstamps, timestamps), Clustering adds 1 (`tx.clock()`). We are going to converge on this, but now is not the time - I need this to land because it is delaying progress, and has been since the 1.0 release. This is a blocker and we can't just converge now, because I need this to be testable.
2. Async calls in futures: that is the solution, but that is a lot of overhead when the alternative is sized clocks. Async futures will land in rust soon (already in nightly, destined for a December stable).